encoders/src/main/java/com/google/firebase/encoders/json/JsonValueObjectEncoderContext.java

```diff
@@ -0,0 +1,175 @@
+      @NonNull Map<Class, ValueEncoder> defaultExtendedEncoders) {
+    this.writer = writer;
+    this.defaultEncoders = defaultEncoders;
+    this.defaultExtendedEncoders = defaultExtendedEncoders;
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(@NonNull String name, @Nullable Object o)
+      throws IOException, EncodingException {
+    jsonWriter.name(name);
+    if (o == null) {
+      jsonWriter.nullValue();
+      return this;
+    }
+    if (o.getClass().isArray()) {
+      jsonWriter.beginArray();
+      if (o.getClass().getComponentType() == int.class) {
+        int[] array = (int[]) o;
+        for (int item : array) {
+          jsonWriter.value(item);
+        }
+      } else if (o.getClass().getComponentType() == double.class) {
+        double[] array = (double[]) o;
+        for (double item : array) {
+          jsonWriter.value(item);
+        }
+      } else if (o.getClass().getComponentType() == boolean.class) {
+        boolean[] array = (boolean[]) o;
+        for (boolean item : array) {
+          jsonWriter.value(item);
+        }
+      } else {
+        Object[] array = (Object[]) o;
+        for (Object item : array) {
+          addSingle(item);
+        }
+      }
+      jsonWriter.endArray();
+      return this;
+    }
+    if (o instanceof Collection) {
+      Collection collection = (Collection) o;
+      jsonWriter.beginArray();
+      for (Object elem : collection) {
+        addSingle(elem);
+      }
+      jsonWriter.endObject();
+      return this;
+    }
+    return addSingle(o);
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(@NonNull String name, double value)
+      throws IOException, EncodingException {
+    jsonWriter.name(name);
+    return add(value);
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(@NonNull String name, int value)
+      throws IOException, EncodingException {
+    jsonWriter.name(name);
+    return add(value);
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(@NonNull String name, boolean value)
+      throws IOException, EncodingException {
+    jsonWriter.name(name);
+    return add(value);
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(@Nullable String value)
+      throws IOException, EncodingException {
+    jsonWriter.value(value);
+    return this;
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(double value) throws IOException, EncodingException {
+    jsonWriter.value(value);
+    return this;
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(int value) throws IOException, EncodingException {
+    jsonWriter.value(value);
+    return this;
+  }
+
+  @NonNull
+  @Override
+  public JsonValueObjectEncoderContext add(boolean value) throws IOException, EncodingException {
+    jsonWriter.value(value);
+    return this;
+  }
+
+  // Does not support collections
```

Comment:
Do we plan to support collections? if so can we rename it to `add()`?

Reply:
Done.